server/core/runtime/policy/conftest_executor.go

```diff
@@ -62,62 +55,61 @@ func NewConfTestExecutor(creator githubapp.ClientCreator, org string) *ConfTestE
 // Run performs conftest policy tests against changes and fails if any policy does not pass. It also runs an all-or-nothing
 // filter that will filter out all policy failures based on the filter criteria.
 func (c *ConfTestExecutor) Run(_ context.Context, prjCtx command.ProjectContext, executablePath string, envs map[string]string, workdir string, extraArgs []string) (string, error) {
-	var policyArgs []Arg
 	var policyNames []string
 	var failedPolicies []valid.PolicySet
+	var totalCmdOutput string
+	var policyErr error
+
 	inputFile := filepath.Join(workdir, prjCtx.GetShowResultFileName())
 	scope := prjCtx.Scope.SubScope(conftestScope)
 
 	for _, policySet := range prjCtx.PolicySets.PolicySets {
-		path, err := c.SourceResolver.Resolve(policySet)
-		// Let's not fail the whole step because of a single failure. Log and fail silently
-		if err != nil {
-			prjCtx.Log.ErrorContext(prjCtx.RequestCtx, fmt.Sprintf("Error resolving policyset %s. err: %s", policySet.Name, err.Error()))
-			continue
+		var policyArgs []Arg
+		for _, path := range policySet.Paths {
+			policyArgs = append(policyArgs, NewPolicyArg(path))
 		}
-		policyArgs = append(policyArgs, NewPolicyArg(path))
 		policyNames = append(policyNames, policySet.Name)
-	}
-
-	args := ConftestTestCommandArgs{
-		PolicyArgs: policyArgs,
-		ExtraArgs:  extraArgs,
-		InputFile:  inputFile,
-		Command:    executablePath,
-	}
-	serializedArgs, err := args.build()
-	if err != nil {
-		prjCtx.Log.WarnContext(prjCtx.RequestCtx, "No policies have been configured")
-		scope.Counter(metrics.ExecutionErrorMetric).Inc(1)
-		return "", errors.Wrap(err, "building args")
-	}
+		args := ConftestTestCommandArgs{
+			PolicyArgs: policyArgs,
+			ExtraArgs:  extraArgs,
+			InputFile:  inputFile,
+			Command:    executablePath,
+		}
+		serializedArgs, err := args.build()
+		if err != nil {
+			prjCtx.Log.WarnContext(prjCtx.RequestCtx, "No policies have been configured")
+			scope.Counter(metrics.ExecutionErrorMetric).Inc(1)
+			return "", errors.Wrap(err, "building args")
+		}
 
-	// TODO: run each policy set separately and use each pass/failure decision to populate failedPolicies
-	cmdOutput, policyErr := c.Exec.CombinedOutput(serializedArgs, envs, workdir)
-	if policyErr != nil {
-		failedPolicies = prjCtx.PolicySets.PolicySets
+		cmdOutput, cmdErr := c.Exec.CombinedOutput(serializedArgs, envs, workdir)
+		// Continue running other policies if one fails since it might not be the only failing one
+		if cmdErr != nil {
+			policyErr = cmdErr
```

Comment:
what happens when more than one policy fails execution? I see we're returning policyErr at the end, but won't it get overwritten if there is more than one policy failure?

Reply:
Good question, it shouldn't be an issue as multiple policy errors would share the same `ExitError` status (the cmdOutput is where the reason for the failure would be added). I only return the latest one because the caller of the ConftestExecutor (the Policy Check Step runner) just wants to know whether a failure existed. 

I could have stopped and returned at the sign of first failure, but I thought it was better to attempt and share all output to the user if they have multiple policy failures. I also did consider modifying the callers to accept multiple runs of the executor, but that would require more refactoring of the legacy code and I didn't think it was worth it.

Reply:
Gotcha, that makes sense! Thanks for the clarification :)